AngularJS/src/org/angular2/entities/Angular2CtorBoundParameters.java

```diff
@@ -0,0 +1,25 @@
+package org.angular2.entities;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * A simple object which hold constructor's bound parameters.
+ */
+public class Angular2CtorBoundParameters {
```

Comment:
I would try to avoid this object. Inputs and outputs can similarly be defined in constructor params and there is no special object to hold them.